### PR TITLE
Arnej/get memory usage from tensors 

### PR DESCRIPTION
**Pull request by arnej27959**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

@geirst please review
@havardpe please take a look at stash changes
@baldersheim FYI
@toregge FYI

### CODE REVIEW

File: searchlib/src/tests/tensor/direct_tensor_store/direct_tensor_store_test.cpp [hunk 1]
```diff
@@ -61,11 +61,12 @@ TEST_F(DirectTensorStoreTest, heap_allocated_memory_is_tracked)
     store.store_tensor(make_tensor(5));
     auto mem_1 = store.getMemoryUsage();
     auto ref = store.store_tensor(make_tensor(10));
-    size_t tensor_memory_used = store.get_tensor(ref)->count_memory_used();
+    auto tensor_memory_used = store.get_tensor(ref)->get_memory_usage();
```
Comment (geirst) [inline searchlib/src/tests/tensor/direct_tensor_store/direct_tensor_store_test.cpp:64]: Consider rename _tensor_memory_used_ to e.g _tensor_memory_. Applies further down in file as well.

**Review by geirst — APPROVED** [review]
👍

**Review by havardpe — APPROVED** [review]
stash extension looks good

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**